osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacyInputDrum.cs

```diff
@@ -159,8 +159,8 @@ public bool OnPressed(KeyBindingPressEvent<TaikoAction> e)
                     const float up_time = 50;
 
                     target.Animate(
-                        t => t.FadeTo(Math.Min(target.Alpha + alpha_amount, 1), down_time)
-                    ).Then(
+                        t => t.FadeTo(Math.Min(target.Alpha + alpha_amount, 1), down_time, Easing.Out)
```

Comment:
Hmm, this may work okay, but the way it's written is slightly weird. Usually we only chain transforms with `Then` when taking the delay from the previous transform, which you're bypassing.

In addition, I don't think the `FadeTo` was correct. This is probably not noticeable in testing, and will only have any effect with really rapid actuations, but yeah.

I'd propose writing it like this (please check it still matches):

```diff
diff --git a/osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacyInputDrum.cs b/osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacyInputDrum.cs
index d9e94ede4a..7992e166ff 100644
--- a/osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacyInputDrum.cs
+++ b/osu.Game.Rulesets.Taiko/Skinning/Legacy/LegacyInputDrum.cs
@@ -153,16 +153,12 @@ public bool OnPressed(KeyBindingPressEvent<TaikoAction> e)
 
                 if (target != null)
                 {
-                    const float alpha_amount = 1;
-
                     const float down_time = 80;
                     const float up_time = 50;
 
-                    target.Animate(
-                        t => t.FadeTo(Math.Min(target.Alpha + alpha_amount, 1), down_time, Easing.Out)
-                    ).Delay(100).Then(
-                        t => t.FadeOut(up_time)
-                    );
+                    target
+                        .FadeTo(1, down_time * (1 - target.Alpha), Easing.Out)
+                        .Delay(100).FadeOut(up_time);
                 }
 
                 return false;

```

Reply:
I've applied the proposed changes and it still matches 👍